diff -Naur yafc-1.0.orig/src/ftp/ftp.c yafc-1.0/src/ftp/ftp.c
--- yafc-1.0.orig/src/ftp/ftp.c 2004-02-10 10:30:22.065639176 +0100
+++ yafc-1.0/src/ftp/ftp.c  2004-02-10 10:26:24.351777176 +0100
@@ -1372,13 +1372,13 @@
         _failed = (ftp_list("MLSD", asdf, fp) != 0);
         free(asdf);
 #else
-        _failed = (ftp_list("MLSD", 0, fp) != 0);
+        _failed = (ftp_list("MLSD", "-a", fp) != 0);
 #endif
         if(_failed && ftp->code == ctError)
             ftp->has_mlsd_command = false;
     }
     if(!ftp->has_mlsd_command) {
-        _failed = (ftp_list("LIST", 0, fp) != 0);
+        _failed = (ftp_list("LIST", "-a", fp) != 0);
         is_mlsd = false;
     }
 
