diff -ur yafc-1.0_orig/src/get.c yafc-1.0_duck/src/get.c
--- yafc-1.0_orig/src/get.c    2003-12-10 11:03:34.000000000 +0100
+++ yafc-1.0_duck/src/get.c    2003-12-10 12:40:55.000000000 +0100
@@ -98,7 +98,7 @@
 {
     int r = 0;
     char *fulldest;
-    char tmp[PATH_MAX];
+    char *tmp;
     transfer_mode_t type;
 
     type = ascii_transfer(src) ? tmAscii : gvDefaultType;
@@ -107,7 +107,10 @@
     else if(test(opt, GET_BINARY))
         type = tmBinary;
 
-    getcwd(tmp, PATH_MAX);
+    tmp = getcwd(NULL, 0);
+    if (tmp == (char *)NULL)
+        return -1;
+
     fulldest = path_absolute(dest, tmp, 0);
 
 #if 0 && (defined(HAVE_SETPROCTITLE) || defined(linux))
@@ -137,6 +140,9 @@
     if(gvUseEnvString && ftp_connected())
         setproctitle("%s", ftp->url->hostname);
 #endif
+
+    free(tmp);
+
     return r;
 }
 
diff -ur yafc-1.0_orig/src/lglob.c yafc-1.0_duck/src/lglob.c
--- yafc-1.0_orig/src/lglob.c	2003-12-10 11:04:33.000000000 +0100
+++ yafc-1.0_duck/src/lglob.c	2003-12-10 12:50:53.000000000 +0100
@@ -64,7 +64,7 @@
 	struct dirent *de;
 	DIR *dp;
 	char *directory;
-	char tmp[PATH_MAX];
+	char *tmp;
 	bool added = false, found = false;
 
 	directory = base_dir_xptr(mask);
@@ -74,12 +74,9 @@
 		return -1;
 	}
 
-	if (!getcwd(tmp, PATH_MAX)) {
-		if (ERANGE == errno) {
-			ftp_err("cwd too long\n");
-		} else {
-			ftp_err("getcwd(): %s\n", strerror(errno));
-		}
+	tmp = getcwd(NULL, 0);
+	if (tmp == (char *)NULL) {
+		ftp_err("getcwd(): %s\n", strerror(errno));
 		return -1;
 	}
 
diff -ur yafc-1.0_orig/src/libmhe/shortpath.c yafc-1.0_duck/src/libmhe/shortpath.c
--- yafc-1.0_orig/src/libmhe/shortpath.c	2003-12-10 11:03:39.000000000 +0100
+++ yafc-1.0_duck/src/libmhe/shortpath.c	2003-12-10 12:34:05.000000000 +0100
@@ -65,9 +65,10 @@
 
 static char *_shortpath(char *path, unsigned maxlen)
 {
-	static char tmp[PATH_MAX];
+	static char *tmp = NULL;
 	int len = strlen(path);
 
+	tmp = realloc(tmp, len+1);
 	strcpy(tmp, path);
 
 	if(len <= maxlen)
@@ -81,12 +82,14 @@
 char *shortpath(const char *path, unsigned maxlen, const char *home)
 {
 	char *e;
-	static char tmp[PATH_MAX];
+	static char *tmp = NULL;
+	int len = strlen(path);
 	extern bool gvTilde;
 
 	if(!path)
 		return 0;
 
+	tmp = realloc(tmp, len+1);
 	strcpy(tmp, path);
 
 	path_collapse(tmp);
diff -ur yafc-1.0_orig/src/local.c yafc-1.0_duck/src/local.c
--- yafc-1.0_orig/src/local.c	2003-12-10 11:03:38.000000000 +0100
+++ yafc-1.0_duck/src/local.c	2003-12-10 12:50:23.000000000 +0100
@@ -21,7 +21,7 @@
 /* print local working directory */
 void cmd_lpwd(int argc, char **argv)
 {
-	char tmp[PATH_MAX+1];
+	char *tmp;
 
 	if(argv) {
 		OPT_HELP("Print local working directory.  Usage:\n"
@@ -32,7 +32,8 @@
 		maxargs(optind - 1);
 	}
 
-	if(getcwd(tmp, PATH_MAX) == 0) {
+	tmp = getcwd(NULL, 0);
+	if (tmp == (char *)NULL) {
 		fprintf(stderr, _("Couldn't get local working directory...\n"));
 		return;
 	}
@@ -42,7 +43,7 @@
 /* local change directory */
 void cmd_lcd(int argc, char **argv)
 {
-	char *e = 0, tmp[PATH_MAX+1];
+	char *e = 0, *tmp;
 	char *te;
 
 	OPT_HELP("Change local working directory.  Usage:\n"
@@ -63,7 +64,11 @@
 	}
 	if(!e)
 		return;
-	getcwd(tmp, PATH_MAX);
+	tmp = getcwd(NULL, 0);
+	if (tmp == (char *)NULL) {
+		fprintf(stderr, _("Couldn't change local working directory...\n"));
+		return;
+	}
 	te = tilde_expand_home(e, gvLocalHomeDir);
 	if(chdir(te) == -1)
 		perror(te);
diff -ur yafc-1.0_orig/src/prompt.c yafc-1.0_duck/src/prompt.c
--- yafc-1.0_orig/src/prompt.c	2003-12-10 11:03:38.000000000 +0100
+++ yafc-1.0_duck/src/prompt.c	2003-12-10 12:54:40.000000000 +0100
@@ -60,7 +60,7 @@
 	unsigned maxlen;
 	char *ins, *e;
 	bool freeins;
-	char tmp[PATH_MAX+1];
+	char *tmp;
 
 
 	if(!fmt)
@@ -149,12 +149,14 @@
 				ins = shortpath(ftp->curdir, maxlen, ftp->homedir);
 				break;
 			case 'l': /* current local directory */
-				getcwd(tmp, PATH_MAX);
+				tmp = getcwd(NULL, 0);
 				ins = shortpath(tmp, maxlen, 0);
+				free(tmp);
 				break;
 			case 'L': /* basename(%l) */
-				getcwd(tmp, PATH_MAX);
+				tmp = getcwd(NULL, 0);
 				ins = (char *)base_name_ptr(tmp);
+				free(tmp);
 				break;
 			case '%': /* percent sign */
 				ins = "%";
diff -ur yafc-1.0_orig/src/syshdr.h yafc-1.0_duck/src/syshdr.h
--- yafc-1.0_orig/src/syshdr.h	2003-12-10 11:03:34.000000000 +0100
+++ yafc-1.0_duck/src/syshdr.h	2003-12-10 12:28:47.000000000 +0100
@@ -238,8 +238,6 @@
 # endif
 #endif
 
-#include <pathmax.h>  /* define PATH_MAX somehow, requires sys/types.h */
-
 typedef RETSIGTYPE (*sighandler_t)(int);
 
 #ifndef bool
diff -ur yafc-1.0_orig/src/utils.c yafc-1.0_duck/src/utils.c
--- yafc-1.0_orig/src/utils.c	2003-12-10 11:04:33.000000000 +0100
+++ yafc-1.0_duck/src/utils.c	2003-12-10 12:58:15.000000000 +0100
@@ -190,8 +190,12 @@
 
 char *get_local_curdir(void)
 {
-	static char buf[PATH_MAX+1];
-	getcwd(buf, PATH_MAX);
+	static char *buf = NULL;
+
+	if (buf != (char *)NULL)
+		free(buf);
+	buf = getcwd (NULL, 0);
+
 	return buf;
 }
 
